Tests/NetworkServiceTests/NetworkServiceTests.swift

```diff
@@ -1,114 +1,90 @@
-@testable import NetworkService
 import XCTest
+@testable import NetworkService
 
 final class NetworkServiceTests: XCTestCase {
-    struct MockEndpoint: HTTPAPIEndpoint {
-        static var networkEnvironment: String = "test"
-        var baseURL: URL = .init(string: "https://api.example.com")!
-        var path: String = "/endpoint"
-        var task: HTTPTask = .request
-        var method: HTTPMethod = .get
-        var headers: HTTPHeaders = [:]
-    }
-
+    
     var networkService: NetworkService<MockEndpoint>!
-
+    
     override func setUpWithError() throws {
         networkService = NetworkService()
         URLProtocol.registerClass(MockURLProtocol.self)
     }
-
+    
     override func tearDownWithError() throws {
         URLProtocol.unregisterClass(MockURLProtocol.self)
         networkService = nil
     }
-
+    
     func testSuccessfulRequestReturnsExpectedData() async throws {
         // Given
-        let expectedData = "Mocked Response Data".data(using: .utf8)!
-        MockURLProtocol.requestHandler = { request in
-            let response = HTTPURLResponse(url: request.url!, statusCode: 200, httpVersion: nil, headerFields: nil)!
+        let expectedData = try XCTUnwrap("Mocked Response Data".data(using: .utf8))
+        MockURLProtocol.requestHandler = { _ in
```

Comment:
Why do you think that is a good idea not to use a data from the request?

Reply:
Fixed with de3f65c